docs/adr/0006-treat-rollouts-differently.md

```diff
@@ -0,0 +1,31 @@
+# Opting out of studies should not affect rollouts
+
+> Status: Open
+
+> Author: Travis Long
+
+> Deciders: Nimbus team, Trust & Legal
+
+> Date: H1 2022
+
+## Context and Problem Statement
+
+Rollouts are a feature of Nimbus and are distinct from experiments in that they are not measurement instruments but instead represent off-train configuration updates.
+
+Rollouts can serve many purposes:
+- Launching a winning branch of an experiment faster than the trains.
+- Launching a configuration to non-experiment users during an experiment after a short period of verification
+- Configuring different settings for a feature for different audiences remotely
+- A "kill switch" if you want to launch a feature but then turn it off if something goes wrong.
+
+Currently, rollouts are ignored if the users has opted out of studies because Nimbus treats rollouts the same as studies, obeying the user preference settings for experiments.
+
+## Decision Outcome
+
+Because of the differences between rollouts and experiments, the Nimbus team would like to treat them independently from each other.
+
+This would include:
+- Disconnecting rollouts from the "studies" application preference so that rollouts could still be delivered to users who have otherwise opted out of participating in experiments.
+- To preserve user-choice, we will require a new preference in applications that integrate Nimbus. This preference will enable the user to opt-out of rollouts, if desired. The status of this preference will be communicated to Nimbus at initialization and if it changes during runtime.
```

Comment:
Is this necessary? The user-choice at the moment is conflated with experiments. By adding an extra layer of user choice we are increasing our support costs.

Reply:
I don't know if it is necessary or not. From a user's perspective I would say "yes, I would like to have the choice to determine when changes are made", but from an operational standpoint I can see where not having a preference would be simpler.